src/commands/color.ts

```diff
@@ -0,0 +1,31 @@
+import { Client, Message } from 'discord.js';
+
+exports.run = (client: Client, msg: Message, args: string[]) => {
+  const roleName = 'USER-' + msg.author.id;
+  const myRole = msg.guild.roles.find((role) => role.name === roleName);
+  let chosenColor = args[0].toUpperCase();
+
+  if (!/(^#?[0-9A-F]{6}$)|(^#?[0-9A-F]{3}$)/i.test(chosenColor)) { // Ensures first argument is a valid hex color
+    msg.channel.send('Please input a valid hex color code!'); // Output error message to discord channel
+    return;
+  }
+
+  if (/(^#?[0-9A-F]{3}$)/i.test(chosenColor)) { // Tests if chosenColor is a 3 character code
+    // tslint:disable-next-line: max-line-length
+    chosenColor = chosenColor.replace(/#?([0-9A-F])([0-9A-F])([0-9A-F])/, '#$1$1$2$2$3$3'); // Convert 3 character hex codes to 6 characters
+  }
+
+  if (!myRole) {
+    msg.guild.createRole({ // Creates new role with user selected color
+      color: chosenColor,
+      name: roleName,
+      // tslint:disable-next-line: max-line-length
+    }).then(() => msg.member.addRole(msg.guild.roles.find((role) => role.name === roleName)).catch()); // Assigns created role to user
```

Comment:
If we are hitting the character limit for this ts-lint rule too often we can up it to 160 or 175. Or we could also make this more vertical as well for readability. Or break it out into several lines with const declared, which won't really affect performance since gc will collect them all at the end of the arrow function anyway.

Whatever works, it doesn't really matter all that much tbh